plutus-benchmark/bls-benchmarks/src/PlutusBenchmark/BLS12_381/Common.hs

```diff
@@ -0,0 +1,349 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE ViewPatterns      #-}
+
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+{- | Approximations of the sort of computations involving BLS12-381 primitives
+ that one might wish to perform on the chain.  Real on-chain code will have
+ extra overhead, but these examples help to give us an idea of the sort of
+ computation that can feasibly be carried out within the validation budget
+ limits. -}
+module PlutusBenchmark.BLS12_381.Common ( UProg
+                                        , UTerm
+                                        , checkGroth16Verify_Haskell
+                                        , listOfSizedByteStrings
+                                        , mkGroth16VerifyScript
+                                        , mkHashAndAddG1Script
+                                        , mkHashAndAddG2Script
+                                        , mkPairingScript
+                                        , mkUncompressAndAddG1Script
+                                        , mkUncompressAndAddG2Script
+                                        , toAnonDeBruijnProg
+                                        )
+where
+import PlutusCore (DefaultFun, DefaultUni)
+import PlutusTx qualified as Tx
+import UntypedPlutusCore qualified as UPLC
+
+import PlutusTx.Prelude as Tx hiding (sort, (*))
+
+import Data.ByteString (ByteString)
+import Data.ByteString qualified as BS
+import Data.Word (Word8)
+import Hedgehog.Internal.Gen qualified as G
+import Hedgehog.Internal.Range qualified as R
+import System.IO.Unsafe (unsafePerformIO)
+import Text.Printf (printf)
+
+import Prelude (IO, fromIntegral, show)
+
+-------------------------------- PLC stuff--------------------------------
+
+type UTerm   = UPLC.Term    UPLC.NamedDeBruijn DefaultUni DefaultFun ()
+type UProg   = UPLC.Program UPLC.NamedDeBruijn DefaultUni DefaultFun ()
+type UDBProg = UPLC.Program UPLC.DeBruijn      DefaultUni DefaultFun ()
+
+
+compiledCodeToTerm
```

Comment:
dupe? I think a bunch of this stuff is elsewhere or even in `Common`

Reply:
I think there's stuff like that all over the place.  I think people want slightly different things in different places and it's easier just to make up a type every time you need one.  Typing the full types is a pain.